app/javascript/entrypoints/application.js

```diff
@@ -12,24 +16,4 @@ Turbo.start();
 // Example: Import a stylesheet in app/frontend/index.css
 import "../../assets/stylesheets/application.tailwind.css";
 
-addEventListener("turbo:before-render", (event) => {
-  if (document.startViewTransition) {
-    event.preventDefault();
-
-    document.startViewTransition(() => {
-      event.detail.resume();
```

Comment:
Have you seen any weird race conditions with this approach? 

I _think_ the Turbo rendering process is asynchronous, and `event.detail.resume()` just tells Turbo to resume rendering, so there's possibility that the View Transitions API captures the markup before Turbo has render the new page.

It might be preferable to return a promise in the `startViewTransition` callback that resolves on `turbo:render`

Reply:
I don't think I could spot some weird race condition but  this is pretty new so my testing hasn't been extensive